specs/simple-serialize.md

```diff
@@ -325,6 +325,7 @@ Instantiate a container with the full set of deserialized data, matching each me
 |:------------------------------------------|:----------------------------------------------------------------|
 | rawbytes has enough left for length       | ``len(rawbytes) > current_index + LENGTH_BYTES``                |
 | list is not greater than serialized bytes | ``len(rawbytes) > current_index + LENGTH_BYTES + total_length`` |
+| item index is at end of serialized bytes  | ``item_index == current_index + LENGTH_BYTES + total_length``   |
```

Comment:
Maybe also add something like

If the container has a fixed number of elements, verify that the number of items read matches with the expected number of elements. 

It may be a design-choice to allow extending types, and that a decoder which receives a type which has more fields than the definition should be forward-compatible? If so, that should clearly be specified.

Reply:
> If the container has a fixed number of elements, verify that the number of items read matches with the expected number of elements.
This can be only verified with item index? Let's say if the container has 2 expected elements and the raw_bytes has more, the former reads only 2 elements and is ignorant of whether the later has 1 or 2 more elements.

> A design-choice to allow extending types, and that a decoder which receives a type which has more fields than the definition should be forward-compatible?
cc @mratsim, would like to consult your expertise on this.

Reply:
I agree with 

> If the container has a fixed number of elements, verify that the number of items read matches with the expected number of elements.

but without the constraint on containers with a fixed number of elements (as all containers have a fixed number of elements).

> A design-choice to allow extending types, and that a decoder which receives a type which has more fields than the definition should be forward-compatible?

I don't think this intended in the current version of the spec, nor do I think it would be a good idea (even though I'd support some kind of versioning, but only if it's done explicitly).

Reply:
> but without the constraint on containers with a fixed number of elements (as all containers have a fixed number of elements).

So, that's another question I have. Isn't this datatype also meant to encapsulate e.g. a list of transactions? That is, a non-fixed-length list of items which are dynamic-length (length-prefixed).

Because, 
- Either the docs should be updated to indicate that `containers` are used for both "variable-length lists of variable-length types` and also `fixed-length lists of arbitrary types`
- Or, the ssz spec needs another section to explain how to encode/decode such lists, as they most definitely will be needed, imo.

Reply:
Oh, I see. I was misled by this comment on the page you link to:

> 1, Get the number of raw bytes to serialize: it is len(list) * sizeof(element)

Because that implies that all `element`s are of identical fixed size. So that's wrong then?

Reply:
Yes, that must be wrong. There's an [issue](https://github.com/ethereum/eth2.0-specs/issues/232) for that, but it hasn't been fixed yet. I opened a [PR](https://github.com/ethereum/eth2.0-specs/pull/484) for this yesterday.